safe_transaction_service/history/serializers.py

```diff
@@ -406,14 +410,18 @@ def validate_delegator_signature(
     ) -> bool:
         ethereum_client = EthereumClientProvider()
         chain_id = ethereum_client.get_chain_id()
+
         # Accept a message with the current topt and the previous totp (to prevent replay attacks)
-        for previous_totp in (True, False):
+        for previous_totp, chain_id in list(
+            itertools.product((True, False), (chain_id, None))
+        ):
```

Comment:
Since we now consider the `chain_id` to be an optional part of the payload to be signed, we do the cross product between the `chain_id` and the `previous_totp`. So we validate the signature against:
- Chain ID, True (previous totp)
- Chain ID, False
- No Chain ID, True
- No Chain ID, False